Document alarm_control_panel triggers, conditions & actions (#44979)
home-assistant/home-assistant.github.io: source/_integrations/alarm_control_panel.markdown

- **Armed away**: full protection for when nobody is home. All sensors (perimeter and interior) are active.
- **Armed night**: similar to home mode, but tuned for sleeping. Typically covers perimeter sensors and selected interior zones while leaving bedroom areas free.
- **Armed vacation**: extended away protection for longer trips. Some systems enable additional monitoring or alerts in this mode.
- **Armed custom bypass**: armed with one or more zones deliberately skipped. Useful when you want to leave a specific door or window open while arming the rest of the system.
- **Arming**: the alarm is counting down its exit delay. You have a short window to leave the house before monitoring begins.
- **Pending**: the alarm detected a sensor trip and is counting down its entry delay. You have a short window to disarm before the alarm triggers.
- **Triggered**: the alarm has gone off. Sirens, notifications, and any other alert actions are active.
- **Disarming**: the alarm is in the process of being disarmed.
- **Unavailable**: the entity is currently unavailable.
- **Unknown**: the state is not yet known.

> ## Code requirements

Reviewer: ```suggestion
## Using an alarm PIN code in actions
```

> Some alarm panels require a PIN code to arm, disarm, or both. Others work without a code at all. When you use the arm or disarm actions, you pass the code as an optional field. If your panel requires a code and you don't provide one (or provide the wrong one), the action fails silently and the alarm stays in its current state. Check your alarm integration's documentation for details on how codes are handled.

Reviewer: ```suggestion
Some alarm panels require a PIN code to arm, disarm, or both. Others work without a PIN code. If your panel requires a PIN and you omit it or enter the wrong one, the action fails silently, and the alarm stays in its current state.

1. Check your alarm integration's documentation to find out whether a PIN is required for arming, disarming, or both.
2. In your automation or script, add the alarm action.
3. If a PIN is needed, under **Data**, add `code` with your PIN:

      ```yaml
      actions:
        - action: alarm_control_panel.alarm_disarm
          target:
            entity_id: alarm_control_panel.home_alarm
          data:
            code: "1234"
      ```
  ```

> ## The "Changed by" attribute

Reviewer: ```suggestion
## About the "Changed by" attribute
```

If your alarm panel supports it, the **Changed by** (`changed_by`) attribute tells you _who_ last changed the alarm state. This is the name or identifier reported by your alarm system, for example a person's name like "Franck Nijhof" or a key fob ID. You can use this attribute in notifications and automations to keep your household informed about who armed or disarmed the alarm.

> ## What you can do with alarm automations

Reviewer: ```suggestion
## Alarm automation examples
```

The real power of this integration is combining triggers and conditions into automations that protect your home without you having to think about it. Here are a few ideas to get you started.

{% include docs/paste_yaml_tip.md %}

> ### Arm the alarm when everyone leaves

Reviewer: ```suggestion
### Automation: arm the alarm when everyone leaves
```
should we indicate in the title that this is an automation example? Otherwise, it reads like a task/procedure title, and I expect numbered step-by-step instructions.

When the last person leaves the house, arm the alarm in away mode automatically. No more wondering whether you remembered to set the alarm after rushing out the door.

- **Trigger**: Everyone leaves the home zone
- **Action**: Arm the alarm in away mode

{% details "YAML example for arming when everyone leaves" %}

{% example %}
automation: |
  alias: "Arm alarm when everyone leaves"
  triggers:
    - trigger: zone
      entity_id: person.alex
      zone: zone.home
      event: leave
    - trigger: zone
      entity_id: person.jamie
      zone: zone.home
      event: leave
  conditions:
    - condition: not
      conditions:
        - condition: zone
          entity_id:
            - person.alex
            - person.jamie
          zone: zone.home
  actions:
    - action: alarm_control_panel.alarm_arm_away
      target:
        entity_id: alarm_control_panel.home_alarm
{% endexample %}

{% enddetails %}

> ### Disarm when you arrive home

Reviewer: ```suggestion
### Automation: disarm when you arrive home
```

When you pull into the driveway, Home Assistant recognizes you are home and disarms the alarm. You walk through the front door without fumbling for a keypad or app.

- **Trigger**: Your person entity enters the home zone
- **Action**: Disarm the alarm

{% details "YAML example for disarming on arrival" %}

{% example %}
automation: |
  alias: "Disarm alarm on arrival"
  triggers:
    - trigger: zone
      entity_id: person.alex
      zone: zone.home
      event: enter
  actions:
    - action: alarm_control_panel.alarm_disarm
      target:
        entity_id: alarm_control_panel.home_alarm
      data:
        code: "1234"
{% endexample %}

{% enddetails %}

> ### Send an urgent notification when the alarm triggers

Reviewer: ```suggestion
### Automation: send an urgent notification when the alarm triggers
```

- **Pending**: the alarm detected a sensor trip and is counting down its entry delay. You have a short window to disarm before the alarm triggers.
- **Triggered**: the alarm has gone off. Sirens, notifications, and any other alert actions are active.
- **Disarming**: the alarm is in the process of being disarmed.
- **Unavailable**: the entity is currently unavailable.
- **Unknown**: the state is not yet known.

## Using an alarm PIN code in actions

Some alarm panels require a PIN code to arm, disarm, or both. Others work without a PIN code. If your panel requires a PIN and you omit it or enter the wrong one, the action fails silently, and the alarm stays in its current state.

1. Check your alarm integration's documentation to find out whether a PIN is required for arming, disarming, or both.
2. In your automation or script, add the alarm action.
3. If a PIN is needed, under **Data**, add `code` with your PIN:

    ```yaml
    actions:
      - action: alarm_control_panel.alarm_disarm
        target:
          entity_id: alarm_control_panel.home_alarm
        data:
          code: "1234"

>     ```

Reviewer 2: De-indent to match paragraph:

````suggestion
3. If a PIN is needed, under **Data**, add `code` with your PIN:

   ```yaml
   actions:
     - action: alarm_control_panel.alarm_disarm
       target:
         entity_id: alarm_control_panel.home_alarm
       data:
         code: "1234"
   ```
````
